### PR TITLE
[Auth] Error codes

### PR DESCRIPTION
**Pull request by flypaper0**

### CODE REVIEW

File: auth/codes.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Codes
+
+## REJECTED
+
+```sh
+case .userRejectedRequest return 4001
```
Comment (Elyniss) [inline auth/codes.md]: How about we use `1` before Auth specific codes?
```suggestion
case .userRejectedRequest return 14001
```

File: auth/codes.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+```
+
+## VALIDATION
+
+```sh
+case .malformedResponseParams return 7001
+case .malformedRequestParams return 7002
+case .messageCompromised return 7003
+case .messageVerificationFailed return 7004
```
Comment (Elyniss) [inline auth/codes.md]:
```suggestion
case .malformedResponseParams return 12001
case .malformedRequestParams return 12002
case .messageCompromised return 12003
case .messageVerificationFailed return 12004
```

**Review by Elyniss — APPROVED** [review]

File: auth/codes.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Codes
+
+## REJECTED
+
+```sh
+case .userRejectedRequest return 14001
```
Comment (llbartekll) [inline auth/codes.md]: I would reorder them so 14000s are after 12000s

File: auth/codes.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+```
+
+## VALIDATION
+
+```sh
+case .malformedResponseParams return 12001
+case .malformedRequestParams return 12002
+case .messageCompromised return 12003
+case .messageVerificationFailed return 12004
```
Comment (llbartekll) [inline auth/codes.md]: does `signatureVerificationFailed` sounds better?

File: auth/codes.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+## REJECTED
+
+```sh
+case .userRejectedRequest return 14001
+```
+
+## VALIDATION
+
+```sh
+case .malformedResponseParams return 12001
```
Comment (llbartekll) [inline auth/codes.md]: any reason why 12000s not 11000s ?

File: auth/codes.md (hunk 3 above)
Comment (jakubuid) [inline auth/codes.md]: Why can't we inherit common error codes from sign? `userRejectedRequest` is already there, we could create a shared pool of errors and avoid creating the same errors with just different code.

Comment (Elyniss) [reply auth/codes.md]: I think we should distinguish between sessionRequest and authRequest

Comment (jakubuid) [reply auth/codes.md]: Why? For peers it doesn't matter. For Relay as well. Auth peer doesn't know about Sign peer.

Comment (Elyniss) [reply auth/codes.md]: I don't have a strong opinion on it. I just think it is not necessary to create a shared pool of errors just for one error.

Comment (jakubuid) [reply auth/codes.md]: Yea I agree that for only one error it doesn't make sense. Let's revisit this once more shared error occurs.

File: auth/codes.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Codes
+
+## REJECTED
+
+```sh
+case .userRejectedRequest return 14001
+```
+
+## VALIDATION
```
Comment (llbartekll) [inline auth/codes.md]: still wondering if those errors really need codes and specs. They will be never send to the peer. 🤔

File: auth/codes.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Codes
+
```
Comment (llbartekll) [inline auth/codes.md:2]:
```suggestion
## REASON
```sh
case .userDisconnected: return 6000
```

```

**Pull request merged**